yourbench/pipeline/summarization.py

```diff
@@ -0,0 +1,497 @@
+# summarization.py
+# =============================================================================
+# Author: @sumukshashidhar
+#
+# Module: Summarization Pipeline Stage
+# =============================================================================
+"""
+Summarization Stage
+===================
+
+This module handles the summarization stage of the YourBench pipeline. It takes
+documents (with their raw text) and generates concise yet comprehensive summaries
+for each document. Optionally, it can compute and store corpus-level metrics
+(e.g., BLEU, METEOR, ROUGE, BERTScore) if running in debug mode.
+
+Usage:
+------
+1. Ensure the pipeline configuration has an entry for the `summarization` stage
+   with the desired settings. For example:
+
+   summarization:
+     run: true
+     timeout_seconds: 300
+     source_dataset_name: yourbench_dataset
+     source_subset: ingested_documents
+     output_dataset_name: yourbench_dataset
+     output_subset: summarized_documents
+     output_split: train
+
+2. When the pipeline runs, it loads the target dataset, calls the summarization
+   model(s) to produce summaries, logs intermediate steps, and saves the updated
+   dataset with new columns:
+     - raw_document_summary
+     - document_summary
+     - summarization_model
+     - quality_metrics (optional in debug mode)
+
+Error Handling & Logging:
+-------------------------
+- All errors are logged using `loguru` to `logs/summarization.log`.
+- The stage attempts to proceed with partial data even if some calls fail, never
+  abruptly terminating the pipeline.
+- In debug mode, additional corpus-level metrics are computed and logged.
+
+Important Notes:
+----------------
+- This stage relies on the `run_inference` utility function from yourbench.utils.inference_engine
+  for concurrency, timeouts, and model management.
+- Summaries are extracted from the model's output by parsing <final_summary> XML tags.
+- If no valid summary is found, the pipeline substitutes a fallback string.
+
+See Also:
+---------
+- yourbench.utils.inference_engine for concurrency logic
+- yourbench.utils.dataset_engine for loading/saving dataset
+"""
+
+from concurrent.futures import ThreadPoolExecutor
+from concurrent.futures import TimeoutError as FuturesTimeoutError
+from typing import Any, Dict, List, Optional
+
+import evaluate
```

Comment:
why are you using evaluate? it's a wrapper for an insane amount of deps and will make you lib dep tree quite huge

Reply:
removed